src/main/java/emu/grasscutter/server/packet/recv/HandlerSetPlayerBornDataReq.java

```diff
@@ -91,6 +92,10 @@ public void handle(GameSession session, byte[] header, byte[] payload) throws Ex
 			Grasscutter.getLogger().error("Error creating player object: ", e);
 			session.close();
 		}
-	}
 
+		// Call join event.
+		PlayerJoinEvent event = new PlayerJoinEvent(player); event.call();
```

Comment:
SetPlayerBornReq is when players are created (right after opening cutscene) not when they join, if you want to do the PlayerJoinEvent, when they actually join, it should at the end of Player::onLogin